### PR TITLE
[FLINK-16717][k8s] Use headless service for rpc and blob port

### PR DESCRIPTION
**Pull request by DeliangFan**
## What is the purpose of the change

Use headless service instead of  node port when run flink job on kubernetes for better performance and compatibility。

First, the RPC and blob only need a headless service which could avoid iptables or ipvs forwarding. 

Second，serverless K8S(like AWS EKS) in public cloud does not support Node Port，which cause some problem when running flink On serverless K8S.

## Brief change log

Use headless service instead of  node port when run flink job on kubernetes. 

## Verifying this change

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by DeliangFan** [comment]
@StephanEwen @wangyang0918 would you please take a review, thx!

**Comment by tillrohrmann** [comment]
Thanks for creating this PR @DeliangFan and sorry for my late response. Unfortunately, your changes are no longer needed. In the current master, the community has removed the directory `flink-container`. Instead, the definitions for a K8s cluster setup with Flink can be found [here](https://ci.apache.org/projects/flink/flink-docs-master/ops/deployment/kubernetes.html#appendix). As you can see, it proposes to use headless services.

Consequently, this change is only relevant for the 1.10 release.

File: flink-container/kubernetes/jobmanager-rest-service.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  - name: rest
+    port: 8081
+    nodePort: 30081
+  - name: query
+    port: 6125
+    nodePort: 30025
+  type: NodePort
+  selector:
+    app: flink
+    component: jobmanager
```
Comment (tillrohrmann) [inline flink-container/kubernetes/jobmanager-rest-service.yaml:34]:
```suggestion
    component: job-cluster
```

File: flink-container/kubernetes/jobmanager-rest-service.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+#
+apiVersion: v1
+kind: Service
+metadata:
+  name: flink-jobmanager-rest
```
Comment (tillrohrmann) [inline flink-container/kubernetes/jobmanager-rest-service.yaml:22]:
```suggestion
  name: flink-job-cluster-rest
```

File: flink-container/kubernetes/README.md [hunk 3]
```diff
@@ -37,6 +37,10 @@ At last, you should start the task manager deployment:
 
 `FLINK_IMAGE_NAME=<IMAGE_NAME> FLINK_JOB_PARALLELISM=<PARALLELISM> envsubst < task-manager-deployment.yaml.template | kubectl create -f -`
 
+Optional service, that exposes the jobmanager rest port as public Kubernetes node’s port:
+
+`kubectl create -f jobmanager-rest-service.yaml`
```
Comment (tillrohrmann) [inline flink-container/kubernetes/README.md:42]:
```suggestion
`kubectl create -f job-cluster-rest-service.yaml`
```

**Comment by tillrohrmann** [comment]
Merged via f7a4128634301a09c9f379ae760250ef078213f1

**Pull request closed without merging**